src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/ConstantExpectedAnalyzer.cs

```diff
@@ -0,0 +1,566 @@
+                    return UnmanagedHelper<long>.TryCreate(parameterSymbol, attributeData, int.MinValue, int.MaxValue, out parameter);
+                case SpecialType.System_Int64:
+                    return UnmanagedHelper<long>.TryCreate(parameterSymbol, attributeData, long.MinValue, long.MaxValue, out parameter);
+                case SpecialType.System_IntPtr:
+                    return UnmanagedHelper<long>.TryCreate(parameterSymbol, attributeData, int.MinValue, int.MaxValue, out parameter);
+                case SpecialType.System_Single:
+                    return UnmanagedHelper<float>.TryCreate(parameterSymbol, attributeData, float.MinValue, float.MaxValue, out parameter);
+                case SpecialType.System_Double:
+                    return UnmanagedHelper<double>.TryCreate(parameterSymbol, attributeData, double.MinValue, double.MaxValue, out parameter);
+                case SpecialType.System_Boolean:
+                    return UnmanagedHelper<bool>.TryCreate(parameterSymbol, attributeData, false, true, out parameter);
+                case SpecialType.System_String:
+                    return StringConstantExpectedParameter.TryCreate(parameterSymbol, attributeData, out parameter);
+                default:
+                    parameter = null;
+                    return false;
+            }
+        }
+
+        private bool ValidateConstantExpectedParameter(IParameterSymbol parameterSymbol, out ImmutableArray<Diagnostic> diagnostics)
+        {
+            if (!TryGetConstantExpectedAttributeData(parameterSymbol, out var attributeData))
+            {
+                diagnostics = ImmutableArray<Diagnostic>.Empty;
+                return false;
+            }
+
+            switch (parameterSymbol.Type.SpecialType)
+            {
+                case SpecialType.System_Char:
+                    return UnmanagedHelper<char>.Validate(parameterSymbol, attributeData, char.MinValue, char.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_Byte:
+                    return UnmanagedHelper<ulong>.Validate(parameterSymbol, attributeData, byte.MinValue, byte.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_UInt16:
+                    return UnmanagedHelper<ulong>.Validate(parameterSymbol, attributeData, ushort.MinValue, ushort.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_UInt32:
+                    return UnmanagedHelper<ulong>.Validate(parameterSymbol, attributeData, uint.MinValue, uint.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_UInt64:
+                    return UnmanagedHelper<ulong>.Validate(parameterSymbol, attributeData, ulong.MinValue, ulong.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_UIntPtr:
+                    return UnmanagedHelper<ulong>.Validate(parameterSymbol, attributeData, uint.MinValue, uint.MaxValue, Helper, out diagnostics);
```

Comment:
@tannergooding In your opinion, is it a good practice to check for uint.MinValue and uint.MaxValue, since UIntPtr has a platform-dependent size?